diagnostic_aggregator/src/aggregator_node.cpp

```diff
@@ -34,34 +34,37 @@
 
 /**< \author Kevin Watts */
 
-#include <diagnostic_aggregator/aggregator.h>
 #include <exception>
 
-using namespace std;
+#include <rclcpp/executors.hpp>
 
-int main(int argc, char **argv)
+#include "diagnostic_aggregator/aggregator.hpp"
+
+using std::exception;
+
+int main(int argc, char ** argv)
 {
-  ros::init(argc, argv, "diagnostic_aggregator");
-  
-  try
-  {
-  diagnostic_aggregator::Aggregator agg;
-  
-  ros::Rate pub_rate(agg.getPubRate());
-  while (agg.ok())
-  {
-    ros::spinOnce();
-    agg.publishData();
-    pub_rate.sleep();
-  }
-  }
-  catch (exception& e)
-  {
-    ROS_FATAL("Diagnostic aggregator node caught exception. Aborting. %s", e.what());
-    ROS_BREAK();
+  rclcpp::init(argc, argv);
+
+  try {
+    diagnostic_aggregator::Aggregator agg;
+
+    rclcpp::executors::SingleThreadedExecutor exec;
```

Comment:
if only one node is present, there is no need for an executor.
Just call `rclcpp::spin_some(agg)`.